devtools/dev-requirements.txt

```diff
@@ -21,12 +21,12 @@ pre-commit
 # testing and benchmarking
 markupsafe == 2.0.1
 nbmake
-pytest >= 5.0.0
+pytest ~= 8.0.0
```

Comment:
Do we still need it >5.0.0?

Reply:
The `~=` means it must be `>=8.0.0, <=8.1.0`